RecoBTag/DeepFlavour/interface/c_pf_features_converter.h

```diff
@@ -0,0 +1,263 @@
+#ifndef RecoSV_DeepFlavour_c_pf_features_converter_h
+#define RecoSV_DeepFlavour_c_pf_features_converter_h
+
+#include "RecoBTag/DeepFlavour/interface/deep_helpers.h"
+#include "DataFormats/PatCandidates/interface/PackedCandidate.h"
+
+namespace deep {
+
+  // conversion map from quality flags used in PV association and miniAOD one
+  constexpr int qualityMap[8]  = {1,0,1,1,4,4,5,6};
+
+  enum qualityFlagsShiftsAndMasks {
+        assignmentQualityMask = 0x7, assignmentQualityShift = 0,
+        trackHighPurityMask  = 0x8, trackHighPurityShift=3,
+        lostInnerHitsMask = 0x30, lostInnerHitsShift=4,
+        muonFlagsMask = 0x0600, muonFlagsShift=9
+  };
+
+
+  template <typename CandidateType, typename JetType,
+            typename TrackInfoBuilderType,
+            typename ChargedCandidateFeaturesType>
+  void c_pf_packed_features_converter(const CandidateType * c_pf,
+                               const JetType & jet,
+                               const TrackInfoBuilderType & track_info,
+                               const float & drminpfcandsv,
+                               ChargedCandidateFeaturesType & c_pf_features) {
+
+            // track_info has to be built before passing as parameter
+
+            // could be inlined
+            float etasign = 1.;
+            if (jet.eta()<0) etasign =-1.;
+
+            c_pf_features.pt = c_pf->pt();
+            c_pf_features.eta = c_pf->eta();
+            c_pf_features.phi = c_pf->phi();
+            c_pf_features.ptrel = deep::catch_infs_and_bound(c_pf->pt()/jet.pt(),
+                                                             0,-1,0,-1);
+            c_pf_features.erel = deep::catch_infs_and_bound(c_pf->energy()/jet.energy(),
+                                                            0,-1,0,-1);
+            c_pf_features.phirel = deep::catch_infs_and_bound(fabs(reco::deltaPhi(c_pf->phi(),jet.phi())),
+                                                              0,-2,0,-0.5);
```

Comment:
do these parameters make sense?
- the value (the first argument) is by construction between 0 and pi
- -0.5 is the offset
- -2 is the lower bound 
    - lower bound can not be reached. It seems like it is just a magic irrelevant number lower than -0.5.

Reply:
@slava77 what your are saying makes sense, only concern is consistency with traning. @mverzett @jkiesele, what do you think about this?